### PR TITLE
feat(client): commit validation BTM

### PR DESCRIPTION
**Pull request by morhag90**
Close: #7959 

## PR Details

Add commit limit validation in BTM.

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-client/src/VersionControl/Commit.tsx [hunk 1]
```diff
@@ -108,6 +124,24 @@ const Commit = ({ projectId, noChanges }: Props) => {
...
+  const commitChangesData = useCallback(() => {
+    setCommitRunning(true);
+    commit({
+      variables: {
+        message: commitMessage,
+        projectId,
+        bypassLimitations: false,
+      },
+    }).catch(console.error);
+  }, [setCommitRunning, commit, projectId, commitMessage]);
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/VersionControl/Commit.tsx:143]: keep using the existing submit mechanism to execute the commit

File: packages/amplication-client/src/VersionControl/Commit.tsx [hunk 2]
```diff
@@ -156,47 +190,55 @@ const Commit = ({ projectId, noChanges }: Props) => {
...
+                    eventData={{
+                      eventName: AnalyticsEventNames.CommitClicked,
+                    }}
+                    disabled={loading}
+                  >
+                    {noChanges ? "Rebuild" : "Commit changes & build "}
+                  </Button>
+                ) : commitBtnType === CommitBtnType.JumboButton ? (
+                  <JumboButton
+                    text="Generate the code for my new architecture"
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/VersionControl/Commit.tsx:226]: you may need to wrap the JumboButton in another component to get the form from the FormikContext and use form.submit on click

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
@morhag90 looks much simpler now 👍 

please see my notes about the form submit. Let me know if you need more context

File: packages/amplication-client/src/VersionControl/Commit.tsx [hunk 3]
```diff
@@ -156,47 +190,55 @@ const Commit = ({ projectId, noChanges }: Props) => {
...
+                    icon="pending_changes"
+                    onClick={commitChangesData}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/VersionControl/Commit.tsx:228]:
```suggestion
                    onClick={formik.submitForm}
```

File: packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesNextSteps.tsx [hunk 4]
```diff
@@ -92,12 +51,13 @@ export const ApplyChangesNextSteps = ({
...
-          onClick={handleGenerateCodeClicked}
-          text="Generate the code for my new architecture"
-          icon="pending_changes"
-          circleColor={EnumTextColor.ThemeTurquoise}
-        ></JumboButton>
+        <Commit
+          projectId={currentProject.id}
+          noChanges
+          showCommitMessage={false}
+          commitMessage="Generated code with my new architecture"
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesNextSteps.tsx:58]: did @PazYanoverr provided this message?

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesNextSteps.tsx (hunk 4 above)
Comment (PazYanoverr) [reply packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesNextSteps.tsx:58]: No...
@morhag90 - Is it the commit comment when applying the architecture changes?

Comment (morhag90) [reply packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesNextSteps.tsx:58]: @PazYanoverr  It is the commit message that will be shown on the commit page: 
image

Comment (PazYanoverr) [reply packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesNextSteps.tsx:58]: @morhag90 - Can we please update it to "Architecture redesign"?

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**